packages/flutter/lib/src/material/banner.dart

```diff
@@ -270,42 +282,50 @@ class _MaterialBannerState extends State<MaterialBanner> {
       ),
     );
 
+    final double elevation = widget.elevation ?? bannerTheme.elevation ?? 0.0;
     final Color backgroundColor = widget.backgroundColor
         ?? bannerTheme.backgroundColor
         ?? theme.colorScheme.surface;
     final TextStyle? textStyle = widget.contentTextStyle
         ?? bannerTheme.contentTextStyle
         ?? theme.textTheme.bodyText2;
 
-    Widget materialBanner = Container(
-      color: backgroundColor,
-      child: Column(
-        mainAxisSize: MainAxisSize.min,
-        children: <Widget>[
-          Padding(
-            padding: padding,
-            child: Row(
-              children: <Widget>[
-                if (widget.leading != null)
-                  Padding(
-                    padding: leadingPadding,
-                    child: widget.leading,
-                  ),
-                Expanded(
-                  child: DefaultTextStyle(
-                    style: textStyle!,
-                    child: widget.content,
-                  ),
+    Widget materialBanner = SafeArea(
+      top: true,
+      child: Container(
+        margin: EdgeInsets.only(bottom: elevation > 0 ? 10.0 : 0.0),
```

Comment:
Is this necessary? I think it should be left to the scaffold body to inset its own content appropriately

Reply:
This shouldn't affect the Scaffold body actually, its only to allow the Banners shadow to show.

You'll find widgets such as the snackbar doing something [similar](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/snack_bar.dart#L554) as well

Reply:
Same with the SafeArea insets, the MaterialBanner should be the one to handle that. The AppBar is a [perfect example](https://github.com/flutter/flutter/blob/ffb2ecea52/packages/flutter/lib/src/material/app_bar.dart#L1082) for this.